File: File search script.cpp
#include <iostream>
#include <filesystem>
#include <vector>
#include <string>
#include <algorithm>

std::vector<std::string> buscar_archivo(const std::string &directorio, const std::string &nombre_archivo)
{
    std::vector<std::string> rutas_encontradas;

    for (const std::filesystem::directory_entry &entry : std::filesystem::recursive_directory_iterator(directorio))
    {
        // Archivo y no otro tipo
        if (entry.is_regular_file())
        {
            std::string nombre = entry.path().filename().string();
            
            std::string nombre_lower = nombre;
            std::transform(nombre_lower.begin(), nombre_lower.end(), nombre_lower.begin(), ::tolower);

            std::string nombre_archivo_lower = nombre_archivo;
            std::transform(nombre_archivo_lower.begin(), nombre_archivo_lower.end(), nombre_archivo_lower.begin(), ::tolower);

            if (nombre_lower.find(nombre_archivo_lower) != std::string::npos)
            {
                rutas_encontradas.push_back(entry.path().string());
            }
        }  
    }

    return rutas_encontradas;
}

int main()
{
    while (true)
    {
        std::string directorio;

        std::cout << "Enter directory: ";

        std::getline(std::cin, directorio);

        // Verificar si el directorio existe
        if (!std::filesystem::exists(directorio) || !std::filesystem::is_directory(directorio))
        {
            std::cout << "Directory not found" << std::endl;

            continue;
        }

        // Solicitar el nombre del archivo (o parte del nombre)
        std::string nombre_archivo;

        std::cout << "Enter file name: ";

        std::getline(std::cin, nombre_archivo);

        std::cout << "------------------------------------" << std::endl;

        // Buscar el archivo en el directorio
        std::vector<std::string> rutas_encontradas = buscar_archivo(directorio, nombre_archivo);

        // Mostrar los resultados
        if (!rutas_encontradas.empty())
        {
            for (const auto &ruta : rutas_encontradas)
            {
                std::cout << ruta << std::endl;
            }
        }
        else
        {
            std::cout << "File not found" << std::endl;
        }

        std::cout << "------------------------------------\n" << std::endl;
    }

    return 0;
}

File: File search script.py
import os

def buscar_archivo(directorio, nombre_archivo):
    rutas_encontradas = []
    
    for raiz, directorios, archivos in os.walk(directorio):
        for archivo in archivos:
            # Búsqueda parcial sin distinguir mayúsculas/minúsculas
            if nombre_archivo.lower() in archivo.lower():
                rutas_encontradas.append(os.path.join(raiz, archivo))
                
    return rutas_encontradas

while True:
    # Directorio de entrada
    while True:
        directorio_val = input("Enter directory: ").strip('"\'')

        if os.path.isdir(directorio_val):
            break

        print("Wrong directory")
    
    # Solicitar el nombre del archivo (o parte del nombre)
    nombre_archivo = input("Enter file name: ")
    
    print("------------------------------------")
    
    # Buscar el archivo en el directorio
    rutas_encontradas = buscar_archivo(directorio_val, nombre_archivo)
    
    # Mostrar los resultados
    if rutas_encontradas:
        for ruta_val in rutas_encontradas:
            print(ruta_val)
    else:
        print("File not found")
    
    print("------------------------------------\n")
